File: frontend/src/services/UserService.js
// src/services/UserService.js
import axios from "axios";

const USER_API_URL = 'http://localhost:8080/user';

export const getUserById = (id) => axios.get(`${USER_API_URL}/${id}`);

export const updateUser = (id, user) => axios.put(`${USER_API_URL}/${id}`, user);

export const deleteUser = (id) => axios.delete(`${USER_API_URL}/${id}`);

export const getAllUsers = () => axios.get(USER_API_URL);

export const searchUsers = (query) => axios.get(`${USER_API_URL}/search`, { params: { query } });

export const greetUser = () => axios.get(`${USER_API_URL}/greet`);

File: backend/src/main/java/dev/com/BloodAid/service/UserService.java
package dev.com.BloodAid.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import dev.com.BloodAid.repo.UserRepo;
import dev.com.BloodAid.models.User;

@Service
public class UserService {

    @Autowired
    UserRepo userRepo;

    public User getUserById(Long id) {
       return userRepo.findById(id).get();
    }
    public User updateUser(Long id, User updatedUser) {
        User user = getUserById(id);
        user.setUserName(updatedUser.getUsername());
        user.setEmail(updatedUser.getEmail());
        user.setPassword(updatedUser.getPassword());
        user.setPhoneNumber(updatedUser.getPhoneNumber());
        user.setLocation(updatedUser.getLocation());
        
        return userRepo.save(user);
    }
    public void deleteUser(Long id) {
        userRepo.deleteById(id);
    }

    public List<User> getAllUsers() {
        return userRepo.findAll();
    }

    public List<User> searchUsers(String query) {
        return userRepo.findByUserNameContainingIgnoreCase(query);
    }

    
}
